src/runtime_src/tools/xclbinutil/KernelUtilities.cxx

```diff
@@ -386,19 +408,44 @@ XclBinUtilities::addKernel(const boost::property_tree::ptree& ptKernel,
     // Create the new PS kernel instance and add it to the vector
     boost::property_tree::ptree ptIPEntry;
     ptIPEntry.put("m_type", "IP_PS_KERNEL");
+    // IP_PS_KERNEL specific data
+    // Note: extended-data is optional, only appliable to fixed ps kernel (-add-kernel)
+    //       not appicable to non-fixed ps kernel (-add-pskernel)
+    // "extended-data" {
+    //   "fundtional": "0"
+    //   "subtype": "1"
+    //   "dpu_kernel_id": "0x101"
+    // }   
+    boost::optional< const boost::property_tree::ptree& > boExtendedData = ptKernel.get_child_optional("extended-data");
+    if (boExtendedData) {
+      const boost::property_tree::ptree& ptExtendedData = boExtendedData.get();
+      if (!ptExtendedData.empty()) {
+        static const std::string NOT_DEFINED = "<not defined>";
```

Comment:
Being that this information is being used in XML code, it would be better to have this be an empty string.
```suggestion
        static const std::string NOT_DEFINED = "";
```

Reply:
change the code as suggested